docs/source-1.0/spec/core/http-traits.rst

```diff
@@ -1157,12 +1158,12 @@ to the `code` set by the :ref:`http-trait`.
 
 .. rubric:: ``httpResponseCode`` is only used on output
 
-``httpResponseCode`` is ignored when resolving the HTTP bindings of an
-operation's input structure. This means that if a structure that contains
-members marked with the ``httpResponseCode`` trait is used as the top-level
-input structure of an operation, then those members are sent as part of the
-:ref:`protocol-specific document <http-protocol-document-payloads>` sent in
-the body of the request.
+``httpResponseCode`` is ignored when resolving the HTTP bindings of any
```

Comment:
This change would be backwards incompatible and potentially break existing models. It's a 180 on the prior posture of ignoring it, to making it an error. It seems like we should have a new validator and make it a warning instead.

Reply:
The blast radius of existing models is probably 0 for input structures with `@httpResponseCode` applied.

Having stronger constraints helps to avoid misusing the trait in future models which is helpful for API developers.

---

The documentation change rewords existing and intended behavior to avoid misunderstandings about input / error structures.